SETUP.md

```diff
@@ -1,330 +1,395 @@
+- **HTTP Honeypot**: Ports 80, 8000, 8001, 8002
+- **HTTPS Honeypot**: Ports 443, 981, 1311, 8243, 8333, 8443, 8448, 8843
+- **gRPC Client**: Connects to backend:8080
+- **Authentication**: Each agent uses unique 64-character hex auth token
 
-You will need to have [bazel](https://bazel.build/) installed. Just grab the latest version from your OS distribution and install it.
+## Deployment Steps
 
-##### Yara-X
+### Development vs Production
 
-Run the following command to install cargo-c
+**Development Deployment:**
+- Uses self-signed certificates
+- Suitable for testing and internal networks
+- Quick setup with IP addresses
 
-```shell
-cargo install cargo-c
-```
-
-Now clone the YARA-X repository:
-
-```shell
-git clone https://github.com/VirusTotal/yara-x.git
-```
-
-Once cargo-c is installed, go to the directory of the YARA-X repository and run the following command as root:
+**Production Deployment:**
+- Uses Let's Encrypt certificates via Cloudflare DNS
+- Requires domain name and Cloudflare API token
+- Provides trusted HTTPS certificates for UI, API, and domain-based agents
+- Includes nginx reverse proxy for SSL termination
+- Automatic HTTP to HTTPS redirect
 
-```shell
-cargo cinstall -p yara-x-capi --release
-```
+### Step 1: Generate Backend Certificates
 
+#### Development Setup
 
+For development or testing environments:
 
-##### Protobuf
+```bash
+# Generate certificates for your backend server
+./scripts/setup-backend-certs.sh <your-server-ip>
 
-Before you can build, you will need to compile some protobuffers by running a script. For this to work, you need protobuf-compiler installed. On Debian and Ubuntu, you do this with:
-
-```shell
-sudo apt-get install protobuf-compiler
+# Example:
+./scripts/setup-backend-certs.sh 8.8.8.8
 ```
 
-### Build the protobufs
+#### Production Setup
 
-The gRPC services are defined in the backend_service.proto. Build these with the following command:
+For production environments with domain names:
 
-```shell
-./compile_proto.sh
+```bash
+# First, configure Cloudflare API credentials in .env.backend
+nano .env.backend
 ```
 
-There is no output on the terminal unless something went wrong.
-
-### Build the backend
+Add these production settings:
+```env
+# Cloudflare API Configuration (for Let's Encrypt certificates)
+CLOUDFLARE_API_TOKEN=your-cloudflare-api-token
+LOPHIID_CERT_DOMAIN=your-domain.com
+LOPHIID_CERT_EMAIL=your-email@domain.com
+```
 
-To build the backend, run the following command:
+Then generate certificates with production mode:
+```bash
+# Generate production certificates with Let's Encrypt
+./scripts/setup-backend-certs.sh your-domain.com production
 
-```shell
-bazel build //cmd/backend:backend
+# Example:
+./scripts/setup-backend-certs.sh honeypot.mydomain.com production
 ```
 
-This will take care of downloading all the remaining dependencies and build the backend.
+**Cloudflare Setup Requirements:**
+1. Domain managed by Cloudflare DNS
```

Comment:
Ai, I just read up on this. Cloudflare seems to block bots and does other stuff. We actually don't want that.   Shall we update this to https://porkbun.com/ ? Or do you know an alternative ? Ideally really just a DNS provider with an API.